tracer/src/Datadog.Trace/RuntimeMetrics/RuntimeMetricsWriter.cs

```diff
@@ -129,14 +130,41 @@ internal RuntimeMetricsWriter(IDogStatsd statsd, TimeSpan delay, bool inAzureApp
 
         public void Dispose()
         {
+            if (Interlocked.CompareExchange(ref _disposed, 1, 0) == 1)
+            {
+                Log.Debug("Disposing Runtime Metrics but it was already disposed");
+                return;
+            }
+
+            Log.Debug("Disposing Runtime Metrics timer");
+            using (var manualResetEvent = new ManualResetEvent(false))
+            {
+                if (_timer.Dispose(manualResetEvent))
+                {
+                    if (!manualResetEvent.WaitOne(30000))
```

Comment:
I think 30s it's too much, 5s it's enough to push the events.

Reply:
changed to 5s thanks!